beebop/app.py

```diff
@@ -15,16 +34,107 @@ def response_success(data):
     return response
 
 
+def response_failure(error):
+    response = {
+        "status": "failure",
+        "errors": [error],
+        "data": []
+    }
+    return response
+
+
+def check_connection(redis):
+    try:
+        redis.ping()
+        return True
+    except (redis_exceptions.ConnectionError, ConnectionRefusedError):
+        return jsonify(response_failure("Connection to redis failed"))
+
+
 @app.route('/version')
 def report_version():
     """
-    report version of beebop (and poppunk,ska in the future)
+    report version of beebop and poppunk (and ska in the future)
     wrapped in response object
     """
     vers = versions.get_version()
-    response = response_success(vers)
-    response_json = jsonify(response)
-    return response_json
+    return jsonify(response_success(vers))
+
+
+@app.route('/poppunk', methods=['POST'])
+@expects_json(schemas.sketches)
+def run_poppunk():
+    """
+    run poppunks assing_query()
+    input: multiple sketches in json format
+    output: clusters assigned to sketches
+    """
+    sketches = request.json['sketches'].items()
+    project_hash = request.json['projectHash']
+    return run_poppunk_internal(sketches, project_hash, storageLocation, redis)
```

Comment:
At the moment this `redis` is coming globally? We should probably push that into the `app` object, same with `storageLocation`